backend/models/user.models.ts

```diff
@@ -0,0 +1,62 @@
+import {model, Schema} from 'mongoose';
+import bcrypt = require('bcryptjs');
+import jwt = require('jsonwebtoken');
+import { databaseSecret } from '../environment';
```

Comment:
What's this for?

Reply:
Those piece are needed to create the JWT tokens used for authentication of our app. It's a bit involved, but well worth the time if you choose to research it more.